src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Dynamo.Core;
+using Dynamo.Engine;
+using Dynamo.Graph.Nodes;
+using Dynamo.Library;
+using Dynamo.Models;
+using Dynamo.Search;
+using Dynamo.Search.SearchElements;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.Reflection;
+using ProtoCore.Utils;
+using ProtoFFI;
+using static Dynamo.Engine.LibraryServices;
+
+namespace NodeDocumentationMarkdownGenerator
+{
+    internal static class AssemblyHandler
+    {
+        /// <summary>
+        /// Scans a list of assemblies using a Reflection only load context to determine which nodes are in the assemblies
+        /// </summary>
+        /// <param name="assemblyPaths">List of dll paths that should be scanned</param>
+        /// <param name="additionalPathsToLoad">List of dll paths that should be added to the PathAssemblyResolver</param>
```

Comment:
why would someone add paths to the path resolver - please make usage obvious and detailed.